compiler/src/dotty/tools/dotc/transform/PatternMatcher.scala

```diff
@@ -325,15 +325,16 @@ object PatternMatcher {
         def isSyntheticScala2Unapply(sym: Symbol) =
           sym.isAllOf(SyntheticCase) && sym.owner.is(Scala2x)
 
+        def tupleApp(i: Int, receiver: Tree) = // manually inlining the call to NonEmptyTuple#apply, because it's an inline method
```

Comment:
Some trees can be shared, but I think Ident trees aren't.  So let's change this

```suggestion
        def tupleApp(i: Int, receiver: => Tree) = // manually inlining the call to NonEmptyTuple#apply, because it's an inline method
```

Reply:
Why is this necessary? `receiver` is used only once inside the `tupleApp` body, so can it not be computed before it is passed to the function?

Reply:
You're right.